### PR TITLE
New integration: gitlab-fork-compliance

### PR DESCRIPTION
**Pull request by apahim**
This re-implements the `test_fork.sh` as an integration.

### CODE REVIEW

File: reconcile/gitlab_fork_compliance.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+from reconcile import queries
+from utils.gitlab_api import GitLabApi
+from utils.gitlab_api import MRState
+
+
+_LOG = logging.getLogger(__name__)
+
+QONTRACT_INTEGRATION = 'gitlab-fork-compliance'
+
+MAINTAINERS_GROUP = 'app-sre'
```
Comment (maorfr) [inline reconcile/gitlab_fork_compliance.py]: can this be an argument?

File: reconcile/gitlab_fork_compliance.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+BOT_USER = 'devtools-bot'
```
Comment (maorfr) [inline reconcile/gitlab_fork_compliance.py]: can we get this from the gitlab cli?

File: reconcile/gitlab_fork_compliance.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+BLOCKED_LABEL = 'blocked/devtools-bot-access'
```
Comment (maorfr) [inline reconcile/gitlab_fork_compliance.py]: let's take this opportunity to change the label to `blocked/bot-access` (i already added the label in the project)

File: reconcile/gitlab_fork_compliance.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    settings = queries.get_app_interface_settings()
+
+    gitlab_cli = GitLabApi(instance, project_id=project_id,
+                           settings=settings)
+
+    group = gitlab_cli.gl.groups.get(MAINTAINERS_GROUP)
+    maintainers = group.members.list()
+    bot = gitlab_cli.get_user(BOT_USER)
+
+    for mr in gitlab_cli.get_merge_requests(state=MRState.OPENED):
```
Comment (maorfr) [inline reconcile/gitlab_fork_compliance.py]: unlike gitlab-owners, this integration should run as part of a pr check of every MR.
so it should get the merge request id as an argument and only act on that MR.

File: reconcile/gitlab_fork_compliance.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        source_project = GitLabApi(instance, project_id=mr.source_project_id,
+                                   settings=settings)
+        blocked = BLOCKED_LABEL in gitlab_cli.get_merge_request_labels(mr.iid)
+
+        # If the bot is not a maintainer on the fork,
```
Comment (maorfr) [inline reconcile/gitlab_fork_compliance.py]: there is a missing check here - if the merge request's source branch is `master` - we should block it as well:
https://gitlab.cee.redhat.com/service/app-interface/blob/master/hack/test_fork.sh#L9

File: reconcile/gitlab_fork_compliance.py (hunk 1 above)
Comment (apahim) [reply reconcile/gitlab_fork_compliance.py]: sure

File: reconcile/gitlab_fork_compliance.py (hunk 5 above)
Comment (apahim) [reply reconcile/gitlab_fork_compliance.py]: o/ my bad

**Pull request merged**